roles/ceph-common/tasks/installs/install_suse_packages.yml

```diff
@@ -5,10 +5,11 @@
     state: present
   register: result
   until: result is succeeded
+  when: ansible_python_version is version( '3', '<')
```

Comment:
The variable `suse_package_dependencies` is not intended to install `python-xml` only, someone might want install packages which could be needed regardless the installed python version.

Reply:
Then I would suggest to leave this variable empty (for people to use if they need it) and add a task installing python-xml explicitely, but only if python2 was found as python interpreter.

Would that be a valid approach in your opinion?

Reply:
@johanneskastl I don't even know why this package is needed when deploying on Suse, any idea?

Reply:
For python2, this one is needed to have ansible's zypper module work correctly, as far as I remember. Python3 includes this by default, so no need to install anything.

I think the cleanest solution would be to move the installation of python-xml to the raw_install_python.yml, as this would be the correct place to install it, before any of ansible's modules are being used.

This would mean my changes in this PR would be void.

Reply:
See #4351 if this is fine with you, then we could close this PR.

Reply:
in `raw_install_python.yml`, it will install `python-xml` only if python wasn't detected, what if you have python but you don't have `python-xml` installed?

Reply:
given that this `python-xml` is a requirement when python2 only, I think, your first approach :

> Then I would suggest to leave this variable empty (for people to use if they need it) and add a task installing python-xml explicitely, but only if python2 was found as python interpreter.
> 
> Would that be a valid approach in your opinion?

is the best option.

Reply:
As raw_install_python.yml has all the logic for finding the python installation(s), I would suggest to add a separate python-xml task there.

Reply:
You will need to install this package after facts are gathered which happens after `raw_install_python.yml` is called.

Reply:
I tried to tackle this in [commit 7e354de](https://github.com/ceph/ceph-ansible/pull/4351/commits/7e354deeb0be70ee4af6f40d0e843b4e954d5965) in PR #4351